File: Calculator.java
import org.antlr.v4.runtime.ANTLRInputStream;
import org.antlr.v4.runtime.CommonTokenStream;

import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

import static java.lang.System.*;

public class Calculator {

    public static void main(String[] args) throws IOException {
      InputStream is = in;
      if(args.length > 0) is = new FileInputStream(args[0]);

      CalculatorLexer lexer = new CalculatorLexer(new ANTLRInputStream(is));
      CalculatorParser parser = new CalculatorParser(new CommonTokenStream(lexer));

      new CalculatorVisitorImpl().visit(parser.calculator());
    }
}

File: src/ruby/calculator.rb
import 'org.antlr.v4.runtime.ANTLRInputStream'
import 'org.antlr.v4.runtime.CommonTokenStream'
import 'java.io.FileInputStream'
import 'CalculatorLexer'
import 'CalculatorParser'
require 'calculator_visitor_impl'

module Calculator
  extend self

  def eval(input_path)
    input_stream  = FileInputStream.new(input_path);
    input         = ANTLRInputStream.new(input_stream)
    lexer         = CalculatorLexer.new(input)
    token_stream  = CommonTokenStream.new(lexer)
    parser        = CalculatorParser.new(token_stream)
    visitor       = CalculatorVisitorImpl.new
    visitor.visit(parser.calculator)
    visitor.results
  end
end
